### PR TITLE
Abstract function for 'have machines resources backing nodes' story

### PR DESCRIPTION
**Pull request by enxebre**
Abstract function for 'have machines resources backing nodes' story
Enforce same number of nodes and machines Fixed by https://github.com/openshift/cluster-api-actuator-pkg/pull/41/commits/3ed9405d07fc4403103ab822ea7b6b61bb4f4091

### CODE REVIEW

File: pkg/e2e/infra/utils.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			glog.Errorf("Error querying api for nodeList object: %v, retrying...", err)
+			return false, nil
+		}
+
+		glog.Infof("Expecting the same number of machines and nodes, have %d nodes and %d machines", len(nodeList.Items), len(machineList.Items))
+		if len(machineList.Items) != len(nodeList.Items) {
+			return false, nil
+		}
+
+		nodeNameToMachineAnnotation := make(map[string]string)
```
Comment (frobware) [inline pkg/e2e/infra/utils.go]: Why not create a function that does this? Can we / will we do this elsewhere?

File: pkg/e2e/infra/utils.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+				return false, nil
+			}
+			nodeNameToMachineAnnotation[node.Name] = node.Annotations[controllernode.MachineAnnotationKey]
+		}
+		for _, machine := range machineList.Items {
+			if machine.Status.NodeRef == nil {
+				glog.Errorf("Machine %q has no NodeRef, retrying...", machine.Name)
+				return false, nil
+			}
+			nodeName := machine.Status.NodeRef.Name
```
Comment (frobware) [inline pkg/e2e/infra/utils.go]: The temporary doesn't seem to buy us much. Just use `machine.Status.NodeRef.Name`.

File: pkg/e2e/infra/utils.go (hunk 1 above)
Comment (enxebre) [reply pkg/e2e/infra/utils.go]: I'll do later if it happens to be reused

**Comment by enxebre** [comment]
k8s-e2e  fails `machines and nodes, have 5 nodes and 4 machines`

https://ci.openshift.redhat.com/jenkins/job/pull-ci-openshift-cluster-api-actuator-pkg-master-e2e/53/console
@ingvagabund is fixing the expectation to be satisfied by the cluster

Next PR is https://github.com/openshift/cluster-api-actuator-pkg/pull/42

**Pull request merged**